activemodel/lib/active_model/normalization.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+module ActiveModel # :nodoc:
+  module Normalization
+    extend ActiveSupport::Concern
+
+    included do
+      include ActiveModel::Dirty
+      include ActiveModel::Validations::Callbacks
```

Comment:
Why not include `ActiveModel::Attributes` here too (and remove from the examples below)?

On one hand, I do see the value in having a model explicitly declare `include ActiveModel::Attributes`.  But, on the other hand, it's a footgun if `ActiveModel::Normalization` doesn't include all of its dependencies.

Just thinking out loud: I wonder if defining the module as `ActiveModel::Attributes::Normalization` would make it more clear?  Perhaps establishing a convention that including `ActiveModel::Attributes::Foo` also includes `ActiveModel::Attributes`.

Reply:
I wanted this PR to be viable, and I imagined that it'd be most viable if the module could be mixed-in on its own, separate from larger modules.

Having said that, I know that https://github.com/rails/rails/issues/50568 is on the horizon, and it's first TODO is `ActiveModel::Model should include ActiveModel::Attribute by default when it is made public`. If that's the end-goal, I'm less worried about the internal structure of the module hierarchy, and am happy to nest this underneath `ActiveModel::Attributes` and make `ActiveModel::Attributes` a hard requirement.

Reply:
Thinking more on this, if the goal is to merge Attributes into Model:

> ActiveModel::Model should include ActiveModel::Attribute by default when it is made public

I think `ActiveModel::Attributes` should include `ActiveModel::Attributes::Normalization`, then consumers can just include `ActiveModel::Model` to get normalization for free, without requiring knowledge of the nested module.

Does that match your expectations?

Reply:
I think it makes sense to be able to include `ActiveModel::Attributes` without including `Normalization`, so I don't think we should include `Normalization` in `ActiveModel::Attributes`.  However, I would be in favor of including `Normalization` in `ActiveModel::Model`.